src/game.cpp

```diff
@@ -9031,6 +9031,20 @@ std::vector<std::string> game::get_dangerous_tile( const tripoint &dest_loc ) co
 
 bool game::walk_move( const tripoint &dest_loc )
 {
+    if( m.has_flag_ter( "THIN_OBSTACLE", dest_loc ) ) {
```

Comment:
This flag is used in a lot of other places - for example on underbrush. This should be replaced with a new flag.